apps/studio/src/server/modules/collection/__tests__/collection.router.test.ts

```diff
@@ -0,0 +1,910 @@
+        resourceType: "CollectionLink",
+        state: "Published",
+        userId: session.userId,
+      })
+      await setupAdminPermissions({ userId: session.userId, siteId: site.id })
+      const auditSpy = vitest.spyOn(auditService, "logResourceEvent")
+      expect(page.draftBlobId).toBe(null)
+      const originalBlob = await db
+        .transaction()
+        .execute((tx) => getBlobOfResource(tx, page.id))
+
+      const expected = await caller.updateCollectionLink({
+        siteId: site.id,
+        category: "category",
+        ref: "1",
+        linkId: Number(page.id),
+      })
+
+      expect(auditSpy).toHaveBeenCalled()
+      const auditEntry = await db
+        .selectFrom("AuditLog")
+        .where("eventType", "=", "ResourceUpdate")
+        .selectAll()
+        .executeTakeFirstOrThrow()
+      expect(auditEntry.delta.before!).toMatchObject({
+        blob: _.omit(originalBlob, ["createdAt", "updatedAt"]),
+        resource: _.omit(page, ["createdAt", "updatedAt"]),
+      })
+      expect(auditEntry.delta.after!).toMatchObject({
+        blob: _.omit(expected, ["createdAt", "updatedAt"]),
+        resource: _.omit(page, ["createdAt", "updatedAt"]),
+      })
+      expect(auditEntry.userId).toBe(session.userId)
+      const actual = getCollectionItemByPermalink(page.permalink, page.parentId)
+      expect(expected).toMatchObject(actual)
+    })
+    it("should update the collection link successfully", async () => {
+      const { page, site } = await setupPageResource({
+        resourceType: "CollectionLink",
+      })
+      const originalBlob = await db
+        .transaction()
+        .execute((tx) => getBlobOfResource(tx, page.id))
+      await setupAdminPermissions({ userId: session.userId, siteId: site.id })
+      const auditSpy = vitest.spyOn(auditService, "logResourceEvent")
+
+      const expected = await caller.updateCollectionLink({
+        siteId: site.id,
+        category: "category",
+        ref: "1",
+        linkId: Number(page.id),
+      })
+
+      expect(auditSpy).toHaveBeenCalled()
+      const auditEntry = await db
+        .selectFrom("AuditLog")
+        .where("eventType", "=", "ResourceUpdate")
+        .selectAll()
+        .executeTakeFirstOrThrow()
+      expect(auditEntry.delta.before!).toMatchObject({
+        blob: _.omit(originalBlob, ["createdAt", "updatedAt"]),
+        resource: _.omit(page, ["createdAt", "updatedAt"]),
+      })
+      expect(auditEntry.delta.after!).toMatchObject({
+        blob: _.omit(expected, ["createdAt", "updatedAt"]),
+        resource: _.omit(page, ["createdAt", "updatedAt"]),
+      })
+      expect(auditEntry.userId).toBe(session.userId)
+      const actual = getCollectionItemByPermalink(page.permalink, page.parentId)
+      expect(expected).toMatchObject(actual)
+    })
+    it.skip("should fail to update to a deleted `ref`")
+    it.skip("should fail to update to an invalid `ref`")
+  })
+})
+
+// Test util functions
```

Comment:
should we move them into a standalone file to keep the file clean?

Reply:
no strong opinion cos this is test file, will do